### DISCUSSION TITLE
User authentication with 2FA enabled fails

### DISCUSSION THREAD

**Discussion by giacomouberti** [globaleaks/globaleaks-whistleblowing-software#4029] [Q&A]
Hi there everybody, I searched for someone who has the same issue but it looks like it never happened before

Today receivers and admins that has the 2FA enabled on their account couldn't login, while trying access after inputting username and password instead of being asked for the OTP code they got the error "Authentication Failed" even the recovery of the password wasn't successful.

After another admin (without 2FA enabled) removed the 2FA authentication and reset the password of the user they were able to login again. Login was also successful after re-enabling a new 2FA on the user.

I checked logs but I couldn't find any useful insight.
We're running Globaleaks 4.13.14  on Ubuntu 22.04 

Thanks for any suggestion, hope you can help me find the issue and possible solutions

**Comment by evilaliv3** [comment]
Thank you @giacomouberti for reporting this.

Actually i do not think the issue was resolved by disabling and re-enabling the 2fa, it is more likely that for some reason the clock of the system could have had a significant skew impacting the syncronization of the 2FA

**Reply by giacomouberti** [reply]
Thanks @evilaliv3 for your feedback, I thought that it might be the issue as well but what sounds strange is that the users were not even prompted with the requesta of the OTP since the authentincation fails occured after they submit username and password, before imputting the 2FA code.

Do you think it still might be connected with the skef of the clock? I'm just trying to understand how to avoid this issue to occur again in the future. Thanks again for your support

**Reply by evilaliv3** [reply]
Thank you @giacomouberti . Ive no clue at the moment, but we will try to understand this better. Thank you for keeping us posted if it happens again. Based on what you say i consider it is unrelated from the 2FA. Maybe the system is running with few ram and logins were failing for this reason. Releases 4.14.8 is more optimized on this matter.

**Comment by gunslinger66** [comment]
Hi everybody, today I had a similar problem: after enabling 2FA I was able to access as administrator locally on the server but from LAN I got the error "Authentication Failed"  until I changed my password in preferences; after that login from LAN was successful.
